CRUD_Operations.c

```diff
@@ -0,0 +1,173 @@
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#define FILE_NAME "users.txt"
+
+typedef struct {
+    int id;
+    char name[50];
+    int age;
+} User;
+
+//for adding new users
+void create_user(){
+     User user;
+    FILE *file = fopen(FILE_NAME, "a");
+    if (!file) {
+        printf("Error opening file.\n");
+        return;
+    }
+
+    printf("Enter User ID: ");
+    scanf("%d", &user.id);
+    printf("Enter Name: ");
+    scanf(" %[^\n]", user.name);  // we can use this for taking string with spaces
+    printf("Enter Age: ");
+    scanf("%d", &user.age);
+    // adding data to our file
+    fprintf(file, "%d,%s,%d\n", user.id, user.name, user.age);
+    fclose(file);
+
+    printf("Successfully Added\n");
+};
+
+// for just viewing the data
+void read_users(){
+     FILE *file = fopen(FILE_NAME, "r");
+    if (!file) {
+        printf("Error opening file.\n");
+        return;
+    }
+
+    User user;
+    printf("\n**** User Data ****\n");
+    while (fscanf(file, "%d,%49[^,],%d\n", &user.id, user.name, &user.age) != EOF) {
+        printf("ID: %d, Name: %s, Age: %d\n", user.id, user.name, user.age);
+    }
+    fclose(file);
+};
+
+// to change user's data
+void update_user(){
+     int target_id, found = 0;
+    User user;
+    FILE *file = fopen(FILE_NAME, "r");
+    FILE *temp = fopen("temp.txt", "w");
```

Comment:
Why do we need "temp.txt"? Can we perform the same operation without creating new temporary file?

Reply:
Basically, it is used to change or delete the data . In this we are reading the data from users.txt, and then modifing the data if needed into temp file. Then we are replacing users file with temp file.

Also we need this temp file because in C the file size can't be change while running. 
 I don't think so we can directly use this. For this I will go for further studies for another approach.